sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_models.py

```diff
@@ -1283,8 +1291,8 @@ class LifetimeAction(object):
         should be between 1 and 972 (36 * 27).
     """
 
-    def __init__(self, action, lifetime_percentage=None, days_before_expiry=None):
-        # type: (CertificatePolicyAction, Optional[int], Optional[int]) -> None
+    def __init__(self, action=None, lifetime_percentage=None, days_before_expiry=None):
```

Comment:
I would keep the type hint you wrote for this method on the next line, but we probably shouldn't give `action` a default value of `None` even if `None` can be provided. We want to make sure an `action` (`None` or otherwise) still needs to be provided when creating a LifetimeAction.

Reply:
I'm not sure how the `action` works at this stage, I'll leave this question to @chlowell .

Reply:
I don't recall the history of `action` but in general we believed required positional parameters really are required, either by Key Vault or for the client to construct a model. Regardless, we shouldn't change the API to solve a type checking problem. Whether a parameter is required (by Python) is unrelated to the type of its value.